sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/SipRoutingClientBuilder.java

```diff
@@ -0,0 +1,451 @@
+    private RetryPolicy retryPolicy;
+    private RetryOptions retryOptions;
+    private final List<HttpPipelinePolicy> additionalPolicies = new ArrayList<>();
+
+    /**
+     * Sets endpoint of the service
+     *
+     * @param endpoint url of the service
+     * @return The updated {@link SipRoutingClientBuilder} object.
+     * @throws NullPointerException If {@code endpoint} is {@code null}.
+     */
+    @Override
+    public SipRoutingClientBuilder endpoint(String endpoint) {
+        this.endpoint = Objects.requireNonNull(endpoint, "'endpoint' cannot be null.");
+        return this;
+    }
+
+    /**
+     * Sets the {@link HttpPipeline} to use for the service client.
+     *
+     * <p><strong>Note:</strong> It is important to understand the precedence order of the HttpTrait APIs. In
+     * particular, if a {@link HttpPipeline} is specified, this takes precedence over all other APIs in the trait, and
+     * they will be ignored. If no {@link HttpPipeline} is specified, a HTTP pipeline will be constructed internally
+     * based on the settings provided to this trait. Additionally, there may be other APIs in types that implement this
+     * trait that are also ignored if an {@link HttpPipeline} is specified, so please be sure to refer to the
+     * documentation of types that implement this trait to understand the full set of implications.</p>
+     * <p>
+     * If {@code pipeline} is set, all other settings aside from
+     * {@link SipRoutingClientBuilder#endpoint(String) endpoint} are ignored.
+     *
+     * @param pipeline {@link HttpPipeline} to use for sending service requests and receiving responses.
+     * @return The updated {@link SipRoutingClientBuilder} object.
+     */
+    @Override
+    public SipRoutingClientBuilder pipeline(HttpPipeline pipeline) {
+        this.pipeline = pipeline;
+        return this;
+    }
+
+    /**
+     * Sets the {@link HttpClient} to use for sending and receiving requests to and from the service.
+     *
+     * <p><strong>Note:</strong> It is important to understand the precedence order of the HttpTrait APIs. In
+     * particular, if a {@link HttpPipeline} is specified, this takes precedence over all other APIs in the trait, and
+     * they will be ignored. If no {@link HttpPipeline} is specified, a HTTP pipeline will be constructed internally
+     * based on the settings provided to this trait. Additionally, there may be other APIs in types that implement this
+     * trait that are also ignored if an {@link HttpPipeline} is specified, so please be sure to refer to the
+     * documentation of types that implement this trait to understand the full set of implications.</p>
+     *
+     * @param httpClient The {@link HttpClient} to use for requests.
+     * @return The updated {@link SipRoutingClientBuilder} object.
+     */
+    @Override
+    public SipRoutingClientBuilder httpClient(HttpClient httpClient) {
+        this.httpClient = httpClient;
```

Comment:
Same thing as HttpPipeline being set to null after being non-null

Reply:
Log added. https://github.com/Azure/azure-sdk-for-java/pull/26819/commits/55c237da4a6c3d1e1fa3b5492c4bb39270fabf5b